countMVS.py

```diff
@@ -97,483 +94,411 @@ def __str__(self):
-                ls.domain = multiDomainList
+                logging.debug("Log source {} may have multiple domains".format(log_source.sensordeviceid))
+                multidomain_list = get_multiple_domains(db_conn, log_source)
+                log_source.domain = multidomain_list
 
                 # Sanity check that LS does in fact have multiple domains
-                if len(multiDomainList) >= 1:
-                    ls.multidomain = True
+                if len(multidomain_list) >= 1:
+                    log_source.multidomain = True
 
-            if ls.multidomain is False:
-                ls.domain.append("Default")
+            if log_source.multidomain is False:
+                log_source.domain.append("Default")
 
-        elif domainCur.rowcount is 1:
+        elif domain_cursor.rowcount == 1:
             # Only one row returned, this log source is only associated with one domain
-            logging.debug("Log source {} has one domain".format(
-                ls.sensordeviceid))
-            domain = domainCur.fetchone()[0]
-            ls.domain.append(domain)
-            logging.debug("Domain for Log source {} was set to {}".format(
-                ls.sensordeviceid, domain))
-
-    except Exception as e:
-        logging.error('Error retrieving domain for log source {}'.format(
-            ls.sensordeviceid))
-        logging.error(e)
+            logging.debug("Log source {} has one domain".format(log_source.sensordeviceid))
+            domain = domain_cursor.fetchone()[0]
+            log_source.domain.append(domain)
+            logging.debug("Domain for Log source {} was set to {}".format(log_source.sensordeviceid, domain))
+
+    except Exception as domain_error:
+        logging.error('Error retrieving domain for log source {}'.format(log_source.sensordeviceid))
+        logging.error(domain_error)
         return
 
 
-# This command tells us if we're a console.
-def isConsole():
+def is_console():
     try:
-        proc = subprocess.Popen(['/opt/qradar/bin/myver', '-c'],
-                                stdout=subprocess.PIPE,
-                                stdin=subprocess.PIPE,
-                                stderr=subprocess.PIPE)
-        (stdout, stderr) = proc.communicate()
-        consoleFlag = str(stdout)
-        if consoleFlag.rstrip() == "true":
-            return True
-        else:
-            return False
-    except Exception as ex:
-        logging.error("Error determining isConsole")
+        # Can't use popen with resource management in Python 2. If this becomes a Python 3-only script
+        # then rewrite using with, and remove the disable rule.
+        # pylint: disable=consider-using-with
+        proc = subprocess.Popen(['/opt/qradar/bin/myver', '-c'], stdout=subprocess.PIPE)
+        stdout, _ = proc.communicate()
+        return str(stdout).rstrip() == 'true'
+    except Exception:
+        print("Error calling myver -c, assume running on console")
```

Comment:
Not really to do with this merge, but just looking through this has made me think if assuming that it's running on the console is actually the best approach if it fails the `myver` command - might be better to just loudly exit, if the `myver` command fails something bad is going on. I'll chat about this with Smokey.

Reply:
Agreed. Exiting sounds better.